pkg/splunk/enterprise/configuration.go

```diff
@@ -441,6 +441,16 @@ func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaOb
 		return statefulSet, err
 	}
 
+	// add serviceaccount if configured
+	if spec.ServiceAccount != "" {
+		namespacedName := types.NamespacedName{Namespace: statefulSet.GetNamespace(), Name: spec.ServiceAccount}
+		_, err := splctrl.GetServiceAccount(client, namespacedName)
+		if err == nil {
+			// serviceAccount exists
```

Comment:
Do we want to bail here if we have an error?

Reply:
**GetServiceAccount** throws an error as per this [log](https://github.com/splunk/splunk-operator/pull/220/files#diff-b0b10241f20d210433da861923f8978a724ec51c2d73c668b3163951eb3dccedR53).